src/home_robot/home_robot/datasets/spot/data_utils.py

```diff
@@ -0,0 +1,342 @@
+    ).to_matrix()
+    body_tform_vision = frame_helpers.get_a_tform_b(
+        tform_snap, frame_helpers.BODY_FRAME_NAME, frame_helpers.VISION_FRAME_NAME
+    ).to_matrix()
+    cam_to_world = torch.from_numpy(body_tform_vision).inverse() @ torch.from_numpy(
+        body_tform_camera
+    )
+    height = response.shot.image.rows
+    width = response.shot.image.cols
+    camera = PinholeCamera(
+        pos=cam_to_world[:3, 3],
+        orn=cam_to_world[:3, :3],
+        height=height,
+        width=width,
+        fx=response.source.pinhole.intrinsics.focal_length.x,
+        fy=response.source.pinhole.intrinsics.focal_length.y,
+        px=response.source.pinhole.intrinsics.principal_point.x,
+        py=response.source.pinhole.intrinsics.principal_point.y,
+        near_val=None,
+        far_val=None,
+        pose_matrix=None,
+        proj_matrix=None,
+        view_matrix=None,
+        fov=None,
+    )
+
+    # # Later on we can try converting between conventions. Here's a non-working example
+    # from home_robot.utils.camera_view_conventions import convert_camera_view_coordinate_system, CameraViewCoordSystem
+    # SPOT = {'up': '-Y', 'right': '+X', 'forward': '+Z'}
+    # M = convert_camera_view_coordinate_system(M, source_convention=SPOT, output_convention=CameraViewCoordSystem.OPENGL.value).transformedRT
+    # cams = cameras_from_opencv_projection(
+    #     R=M[:, :3, :3], # .permute([0,2,1])
+    #     tvec=M[:, :3, 3],
+    #     # tvec = torch.stack(tvecs).float(),
+    #     camera_matrix=torch.stack(intrinsics).float(),
+    #     image_size=torch.stack(image_sizes).float()
+    # )
+    # # cams = PerspectiveCameras(
+    # #     device="cpu",
+    # #     R=M[:, :3, :3].permute([0,2,1]),
+    # #     T=M[:, :3, 3],
+    # #     K=torch.stack(intrinsics).float())
+    return camera
+
+
+def get_base_gps_compass_from_spot_image_response(response: image_pb2.ImageResponse):
+    raise NotImplementedError
+
+
+def build_obs_from_spot_image_responses(
+    depth_response: image_pb2.ImageResponse,
+    color_response: Optional[image_pb2.ImageResponse],
+):
+    # Get camera parameters
+    camera: PinholeCamera = get_camera_from_spot_image_response(depth_response)
+    cam_to_world = torch.eye(4)
+    cam_to_world[:3, :3] = camera.orn
+    cam_to_world[:3, 3] = camera.pos
+
+    # get images
+    color = None
+    if color_response is not None:
+        color = torch_image_from_spot_image_response(color_response)
+    depth = torch_image_from_spot_image_response(depth_response)
+
+    # Other metadata
+    timestamp = depth_response.shot.acquisition_time.ToDatetime()
+
+    # Build obs
+    return Observations(
+        gps=None,
+        compass=None,
+        rgb=color,
+        depth=depth,
+        camera_pose=cam_to_world,
+        camera_K=camera.K[:3, :3],
+        task_observations={"timestamp": timestamp},
+    )
+
+
+class SpotPickleDataset:
```

Comment:
What script creates this? Maybe a light description? not totally sure, its fairly obvious